cpmpy/transformations/decompose_global.py

```diff
@@ -0,0 +1,78 @@
+import copy
+
+from ..expressions.globalconstraints import GlobalConstraint
+from ..expressions.core import Expression, Comparison, Operator
+from ..expressions.utils import is_any_list, eval_comparison
+from ..expressions.python_builtins import all
+from .flatten_model import flatten_constraint
+
+def decompose_global(lst_of_expr, supported={}):
+
+    """
+        Decomposes any global constraint not supported by the solver
+        Accepts a list of flat constraints as input
+        Returns a list of flat constraints
+    """
+    def _is_supported(cpm_expr):
+        if isinstance(cpm_expr, GlobalConstraint) and cpm_expr.name not in supported:
+            return False
+        if isinstance(cpm_expr, Comparison) and isinstance(cpm_expr.args[0], GlobalConstraint) and cpm_expr.args[0].name not in supported:
+            return False
+        return True
+
+    if not is_any_list(lst_of_expr):
+        lst_of_expr= [lst_of_expr]
+
+    newlist = []
+    for cpm_expr in lst_of_expr:
+        assert isinstance(cpm_expr, Expression), f"Expected CPMpy expression but got {cpm_expr}, run 'cpmpy.transformations.normalize.toplevel_list' first"
+        decomp_idx = None
+
+        if hasattr(cpm_expr, "decompose") and cpm_expr.is_bool() and cpm_expr.name not in supported:
+            cpm_expr = cpm_expr.decompose() # base boolean global constraints
+        elif isinstance(cpm_expr, Comparison):
+            lhs, rhs = cpm_expr.args
+            if cpm_expr.name == "==" and not _is_supported(lhs): # can be both boolean or numerical globals
+                decomp_idx = 0
```

Comment:
how about `_is_supported(lhs)`? Add an assert and comment if this is not possible.

Reply:
What about comparisons that are not inequalities? E.g., `x >= abs(y)`? If this case is not possible, add some assert and comment why.